clang-tools-extra/clang-tidy/bugprone/SizeofExpressionCheck.cpp

```diff
@@ -353,6 +367,19 @@ void SizeofExpressionCheck::check(const MatchFinder::MatchResult &Result) {
     diag(E->getBeginLoc(),
          "suspicious usage of 'sizeof(char*)'; do you mean 'strlen'?")
         << E->getSourceRange();
+  } else if (const auto *E = Result.Nodes.getNodeAs<Stmt>("loop-expr")) {
+    auto *SizeofArgTy = Result.Nodes.getNodeAs<Type>("sizeof-arg-type");
+    if (const auto member = dyn_cast<MemberPointerType>(SizeofArgTy)) {
+      SizeofArgTy = member->getPointeeType().getTypePtr();
+    }
```

Comment:
Please don't use braces for single-line ifs
```suggestion
    if (const auto member = dyn_cast<MemberPointerType>(SizeofArgTy))
      SizeofArgTy = member->getPointeeType().getTypePtr();
```